wp-modules/pattern-data-handlers/pattern-data-handlers.php

```diff
@@ -323,10 +323,7 @@ function update_pattern( $pattern ) {
 function delete_pattern( string $pattern_name ): bool {
 	$wp_filesystem = \PatternManager\GetWpFilesystem\get_wp_filesystem_api();
 	$pattern_path  = get_pattern_path( $pattern_name );
-	$result        = $wp_filesystem && $wp_filesystem->exists( $pattern_path ) && $wp_filesystem->delete( $pattern_path );
-	tree_shake_theme_images( $wp_filesystem, 'copy_dir' );
```

Comment:
I shouldn't have called `tree_shake_theme_images()` in `delete_pattern()`.

It's not needed here. There are 2 other places that call it.

Hiding it in `delete_pattern()` probably caused this bug, and invites further bugs.

Reply:
Like Phil [pointed out](https://github.com/studiopress/pattern-manager/pull/192/commits/d33a28b109f69ff3e079a4cd5aafa180672d5e2d#diff-8ba31aa77fc662d19b7f20a487d3c2e0d83ac9ab99b843479f7ddae96b0649edR147-R1530), this made the order of `delete_pattern()` calls important.

The `rest_after_insert` [callback](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L83) runs after [delete_pattern()](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L151). 

So we shouldn't need to tree-shake in `delete_pattern()`. The `rest_after_insert` callback will do that.

Reply:
The `rest_after_insert` [callback](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L83) runs after [delete_pattern()](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L151). 

So we shouldn't need to tree-shake in `delete_pattern()`. The `rest_after_insert` callback will do that.